checker/src/org/checkerframework/checker/index/upperbound/OffsetEquation.java

```diff
@@ -387,7 +387,7 @@ private static int parseInt(String intLiteral) {
         if (intLiteral.isEmpty()) {
             return 0;
         }
-        return Integer.valueOf(intLiteral);
+        return Long.valueOf(intLiteral).intValue();
```

Comment:
The reason is that array lengths are ints not longs.  I would't change offsets to be longs without a good reason.